2022q4/NFS-in-prisons.adoc

```diff
@@ -1,21 +1,21 @@
 === Enable the NFS server to run in a vnet prison
 
 Links: +
-link:https://people.freebsd.org/~rmacklem/vnet.patch[Source patch for main] URL: link:https://people.freebsd.org/~rmacklem/vnet.patch[https://people.freebsd.org/~rmacklem/vnet.patch] +
-link:https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[Simple Setup Doc] URL: link:https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt]
+https://people.freebsd.org/~rmacklem/vnet.patch[Source patch for main] URL: link:https://people.freebsd.org/~rmacklem/vnet.patch[https://people.freebsd.org/~rmacklem/vnet.patch] +
+https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[Simple Setup Doc] URL: link:https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt]
```

Comment:
I have tested it and it will not work on the FreeBSD website. I have found that the issue is in the ~ symbols in the second occurrence of each link. I have found two possibile solutions:

- we can escape the problematic ~ symbols:

``
link:https://people.freebsd.org/~rmacklem/vnet.patch[Source patch for main] URL: link:https://people.freebsd.org/\~rmacklem/vnet.patch[https://people.freebsd.org/~rmacklem/vnet.patch] +
link:https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[Simple Setup Doc] URL: link:https://people.freebsd.org/\~rmacklem/nfsd-vnet-prison-setup.txt[https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt]
``

- we can URL encode the problematic ~ symbols:

``
link:https://people.freebsd.org/~rmacklem/vnet.patch[Source patch for main] URL: link:https://people.freebsd.org/%7Ermacklem/vnet.patch[https://people.freebsd.org/~rmacklem/vnet.patch] +
link:https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt[Simple Setup Doc] URL: link:https://people.freebsd.org/%7Ermacklem/nfsd-vnet-prison-setup.txt[https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt]
``

I believe it is better to URL encode. We might have some issues in the future with escaping (suppose we switch from adoc to something else), while URL encoding should stay safe.

The fix I propose is the minimal change that makes things work. However it might be a good idea, for consistency, to change other ~ symbols as well. I would go with

``
link:https://people.freebsd.org/%7Ermacklem/vnet.patch[Source patch for main] URL: link:https://people.freebsd.org/%7Ermacklem/vnet.patch[https://people.freebsd.org/~rmacklem/vnet.patch] +
link:https://people.freebsd.org/%7Ermacklem/nfsd-vnet-prison-setup.txt[Simple Setup Doc] URL: link:https://people.freebsd.org/%7Ermacklem/nfsd-vnet-prison-setup.txt[https://people.freebsd.org/~rmacklem/nfsd-vnet-prison-setup.txt]
``

Please note that in my final proposition I have kept the ~ symbols in square brackets for readability. It is possible that keeping them might cause problems in the future, but it is less likely than in the other occurrences and I value readability more. However, feel free to disagree and to URL encode those occurrences of ~ as well if you want to be safer.

Reply:
Thanks. Resolved with [`8adf017` (#573)](https://github.com/freebsd/freebsd-quarterly/pull/573/commits/8adf017f3bada8707eccee3005a18698cc832b1f).